src/mem/PoolMalloc.cc

```diff
@@ -31,6 +31,7 @@ MemPoolMalloc::allocate()
     if (obj) {
         --meter.idle;
         ++countSavedAllocs;
+        (void)VALGRIND_MAKE_MEM_UNDEFINED(obj, objectSize);
```

Comment:
MemPoolChunked::get() uses VALGRIND_MAKE_MEM_DEFINED() instead. I have not checked MemPoolChunked code carefully, but I suspect it is buggy because (non-zeroed) memory bytes returned by a memory pool are, in valgrind [terms](https://valgrind.org/docs/manual/mc-manual.html#mc-manual.clientreqs), accessible but not _defined_ (i.e. we cannot read from that memory until we write to it). This PR does not attempt to address any MemPoolChunked concerns, but it does not copy-paste MemPoolChunked calls either.

FWIW, I would certainly support improving MemPoolChunked memory marking code to address the above concern, but a long-term solution is probably different (as detailed in this PR description).

P.S. This comment does not request any code changes.